### PR TITLE
Updated dependency of onefx to v1.5.2

### LINKED ISSUE #805: bump dependency of onefx to v1.5.2

### PR DESCRIPTION
**Pull request by gyanpgupta**
Fixes #805

### CODE REVIEW

File: src/client/javascripts/main.tsx [hunk 1]
```diff
@@ -28,5 +28,6 @@ clientReactRender({
     signParams: signParamsReducer,
     wallet: walletReducer
   }),
+  // @ts-ignore
```
Comment (puncsky) [inline src/client/javascripts/main.tsx]: don't ignore - just follow the type defs to remove it

File: src/api-gateway/api-gateway.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ import { buildSchema } from "type-graphql";
 import { AntennaResolver } from "./resolvers/antenna";
 import { MetaResolver } from "./resolvers/meta";
 
-export async function setApiGateway(server: Server): Promise<void> {
+// tslint:disable-next-line:no-any
```
Comment (puncsky) [inline src/api-gateway/api-gateway.ts]: add type definition in start-server.ts like 

```
import { Server } from "onefx/lib/server";

type MyServer = Server & {
}
```

and in other places use the type `MyServer`

**Comment by gyanpgupta** [comment]
@guo  updated feedback changes. please have a look thanks

File: src/server/gateways/gateways.ts [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
 import { getSendgrid } from "./get-sendgrid";
 
-export function setGateways(server: Server): void {
+export function setGateways(server: MyServer): void {
   server.gateways = server.gateways || {};
   const gateways = server.config.gateways;
-
   server.gateways.antenna = new RpcMethod(gateways.iotexAntenna);
-  server.gateways.coinmarketcap = { fetchCoinPrice };
+  server.gateways.coinmarketcap = fetchCoinPrice;
```
Comment (puncsky) [inline src/server/gateways/gateways.ts]: Will this break the site?

File: src/shared/wallet/sign-and-send-envelop-modal.tsx [hunk 4]
```diff
@@ -74,6 +74,7 @@ class SignAndSendEnvelopModalInner extends Component<Props> {
     const nonce = String(
       (meta.accountMeta && meta.accountMeta.pendingNonce) || ""
     );
+    // @ts-ignore
```
Comment (puncsky) [inline src/shared/wallet/sign-and-send-envelop-modal.tsx]: Why ignore?

File: src/server/start-server.ts [hunk 5]
```diff
@@ -1,20 +1,51 @@
...
+import { AntennaResolver } from "../api-gateway/resolvers/antenna";
+import { MetaResolver } from "../api-gateway/resolvers/meta";
 import { setModel } from "../model";
 import "../shared/common/setup-big-number";
 import { setGateways } from "./gateways/gateways";
 import { setMiddleware } from "./middleware";
 import { setServerRoutes } from "./server-routes";
 
-export async function startServer(): Promise<void> {
+export type MyServer = Server & {
```
Comment (nmhung1210) [inline src/server/start-server.ts]: It's unnecessary to define `MyServer` type and eventually end up with `ts-ignore` like this. Instead, you can just adapt the `config` to `Config` type by spreading default value like below:

```suggestion
const defaultConfig: Config = {
  project: "",
  server: {
    host: "",
    port: "",
    staticDir: "",
    delayInitMiddleware: false,
    cookie: {
      secrets: []
    },
    noSecurityHeadersRoutes: {},
    noCsrfRoutes: {}
  },
  gateways: {
    logger: {
      enabled: false,
      baseDir: "",
      topicName: "",
      level: "debug"
    }
  },
  csp: {},
  analytics: {},
  session: {}
};

const serverConfig: Config = {
  ...defaultConfig,
  ...config
};

export async function startServer(): Promise<Server> {
  const server = new Server(serverConfig);
  setGateways(server);
  setMiddleware(server);
  setModel(server);
  setServerRoutes(server);

  const port = process.env.PORT || config.get("server.port");

  server.listen(parseInt(`${port}`, 10));
  return server;
}
```

File: src/server/start-server.ts [hunk 6]
```diff
@@ -1,20 +1,51 @@
...
+  vitaTokens: object;
+  multiChain: object;
+  defaultERC20Tokens: object;
+  webBpApiGatewayUrl: string;
+  enableSignIn: boolean;
+  apiGatewayUrl: string;
+};
+
+export async function startServer(): Promise<MyServer> {
+  // @ts-ignore
```
Comment (puncsky) [inline src/server/start-server.ts]: const server = new Server(config as any as MyConfig);

File: src/server/gateways/gateways.ts (hunk 3 above)
Comment (gyanpgupta) [reply src/server/gateways/gateways.ts]: No , I don't think so.

**Pull request merged**